### ISSUE TITLE
Example Code Generating Errors

### ISSUE THREAD

**Issue by eCommerceGorilla** [ParsePlatform/parse-php-sdk#267]
Getting the following error using the provided example code.

Fatal error: Uncaught Error: Wrong parameters for Parse\ParseException([string $message [, long $code [, Throwable $previous = NULL]]]) in /chroot/home/html/parse/src/Parse/ParseException.php:23 Stack trace: #0 /chroot/home/html/parse/src/Parse/ParseException.php(23): Exception->__construct(Array, 0, NULL) #1 /chroot/home/html/parse/src/Parse/ParseClient.php(377): Parse\ParseException->__construct(Array, 0) #2 /chroot/home/html/parse/src/Parse/ParseObject.php(1043): Parse\ParseClient::_request('POST', 'classes/TestObj...', NULL, '{"foo":"bar"}', false) #3 /chroot/home/html/parse/src/Parse/ParseObject.php(954): Parse\ParseObject::deepSave(Object(Parse\ParseObject), false) #4 /chroot/home/html/parse/index.php(30): Parse\ParseObject->save() #5 {main} thrown in /chroot/home/html/parse/src/Parse/ParseException.php on line 23

The code I'm using is the following:
```
require 'autoload.php';

use Parse\ParseObject;
use Parse\ParseQuery;
use Parse\ParseACL;
use Parse\ParsePush;
use Parse\ParseUser;
use Parse\ParseInstallation;
use Parse\ParseException;
use Parse\ParseAnalytics;
use Parse\ParseFile;
use Parse\ParseCloud;
use Parse\ParseClient;

$app_id = "HY......oG";
$rest_key = "hU.........a";
$master_key = "xT..............CY";

ParseClient::initialize( $app_id, $rest_key, $master_key );
// Users of Parse Server will need to point ParseClient at their remote URL and Mount Point:
ParseClient::setServerURL('https://parseapi.back4app.com/','parse');


$object = ParseObject::create("TestObject");

// Signup
$user = new ParseUser();
$user->setUsername("foo");
$user->setPassword("Q2w#4!o)df");
try {
    $user->signUp();
} catch (ParseException $ex) {
    // error in $ex->getMessage();
}

```
It doesn't matter what example code I try it always returns the same. This is true on 4 different hosting accounts I've tested this on including a local AMPPS environment.

**Comment by montymxb** [comment]
It looks like it's passing an Array instead of a string for the Exception constructor.
```
#0 /chroot/home/html/parse/src/Parse/ParseException.php(23): Exception->__construct(Array, 0, NULL)
```
Looking a little further the section below from ```ParseClient.php``` around line 377 seems to be the root of the problem.
```php
$decoded = json_decode($response, true);
if (isset($decoded['error'])) {
	throw new ParseException(
		$decoded['error'],
		isset($decoded['code']) ? $decoded['code'] : 0
	);
}
```

```$decoded['error']``` is assumed to be a string for the first argument. Since there is no checking of this the unexpected type of Array is passing through. In response you're getting the 'Wrong parameters' error. Your error value is probably some JSON instead of the expected string error message.

Assuming the message doesn't contain anything sensitive I can craft together a pull request to perform a check and subsequent json_encode() on the 'error' vaue of ```$decoded```, if you're willing to wait that is. 

If you want to patch this issue in the meantime you could modify the code in ```ParseClient.php``` above to check for and convert an Array to a string as follows:
```php
$decoded = json_decode($response, true);
if (isset($decoded['error'])) {
	$errorMessage = is_array($decoded['error']) ? json_encode($decoded['error']) : $decoded['error'];
	throw new ParseException(
		$errorMessage,
		isset($decoded['code']) ? $decoded['code'] : 0
	);
}
```
Given the stack trace provided that <i>should</i> take care of your error, although you're still going to have to handle the ParseException.

Also I think all of us would be curious as to what back4app.com is actually providing in their error besides the expected string. It's quite possible that other SDKs may (or may not) have a similar issue as well.

**Comment by eCommerceGorilla** [comment]
Thanks for the patch. From the the exception return after patching I was able to determine that it was an invalid end point. What it turns out is to work on their platform I needed to update the stand SDK 

ParseClient::setServerURL('https://parseapi.back4app.com','parse');
to
ParseClient::setServerURL('https://parseapi.back4app.com','/');

I greatly appreciate the assistance. Parse is turning into my favorite out of the box backend by far.
